docs/ChainOrderTransitivity.md

```diff
@@ -0,0 +1,153 @@
+
+## Examples for intransitive orders
+
+### Current chain order is not transitive
+
+For all Shelley-based eras, we currently use a lexicographic combination of the following comparisons in descending order:
+
+ 1. Chain length, preferring longer chains.
+ 2. If the issuer identity is the same, compare by the issue/opcert number, preferring higher values, otherwise, no preference.
+ 3. VRF tiebreaker[^vrf-tpraos-vs-praos], preferring lower values.
+
+This is not a total order: It is is reflexive, antisymmetric and total, but not transitive.
+
+Consider the following three `SelectView`s:
+
+|              | A | B | C |
+| ------------ | - | - | - |
+| Chain length | l | l | l |
+| Issuer       | X | Y | X |
+| Opcert no    | 2 | o | 1 |
+| VRF          | 3 | 2 | 1 |
+
+Lower-case letters stand for arbitrary values (but if they occur repeatedly, they designate the same value).
+
+We have
+
+  - `A < B` due to the VRF tiebreaker,
+  - `B < C` due to the VRF tiebreaker, again, but also
+  - `C < A` as `A` has a higher opcert number than `C`.
+
+So we have `A < B < C < A < B < C < ...`.
+
+### "Duncan's rule" is intransitive
+
+Consider the following alternative rule, for simplicity without the issuer/opcert rule as it is not relevant for the point:
+
+ 1. Chain length, the longer chain is preferred.
+ 2. If the slot numbers differ by at most `Δ = 5` slots, compare the VRF tiebreakers[^vrf-tpraos-vs-praos], preferring lower values, otherwise, no preference.
+
+Again, this is reflexive, antisymmetric and total, but not transitive:
+
+|              | A | B | C |
+| ------------ | - | - | - |
+| Chain length | l | l | l |
+| Slot         | 0 | 3 | 6 |
+| VRF          | 3 | 2 | 1 |
+
+ - `A < B` due to the VRF tiebreaker, as the slots 0 and 3 are within `Δ = 5`.
+ - `B < C` due to the VRF tiebreaker, as the slots 3 and 6 are within `Δ = 5`, again.
+ - `C = A` as the slots 6 and 0 differ by more than `Δ = 5`.
+
+So we have `A < B < C = A < B < C = ...`.
+
+## History
+
+The relevant changes to the chain order occurred in these PRs in chronological order:
+
+ - [ouroboros-network#2108](https://github.com/IntersectMBO/ouroboros-network/pull/2108) added the opcert number comparison step as the first tiebreaker (before that, only chain length was used).
+ - [ouroboros-network#2195](https://github.com/IntersectMBO/ouroboros-network/pull/2195) added the VRF tiebreaker, but lexicographically *before* the opcert number tiebreaker, in contrast to the status quo.
+   This means that the order still was total at this point, also see below.
+ - [ouroboros-network#2348](https://github.com/IntersectMBO/ouroboros-network/pull/2348) mostly did a change unrelated to the purpose of this document, but crucially, it swapped the order in which the VRF and opcert number comparison are done, introducing the current non-transitivity.
+
+## Impact of non-transitivity
```

Comment:
I'd say, if possible say "not transitive". But if that's too awkward, use exactly one of "intransitive" and "non-transitive". Personally I think "non-transitive" is preferable to "intransitive".

Reply:
Yeah, apparently "intransitive" sometimes means "antitransitive": https://en.wikipedia.org/wiki/Intransitivity#Antitransitivity

I think I will use "non-transitive" everywhere, in particular as it has a noun ("non-transitivity") compared to "not transitive" :+1: